src/CommandingOfficers/Modifiers/UnitIndirectRangeModifier.java

```diff
@@ -0,0 +1,28 @@
+package CommandingOfficers.Modifiers;
+
+import Engine.UnitMods.UnitModifierWithDefaults;
+import Units.UnitContext;
+
+public class UnitIndirectRangeModifier implements UnitModifierWithDefaults
+{
+  private static final long serialVersionUID = 1L;
+  private int rangeChange;
+
+  public UnitIndirectRangeModifier()
+  {
+    this(1);
+  }
+
+  public UnitIndirectRangeModifier(int range)
+  {
+    rangeChange = range;
+  }
+
+  @Override
+  public void modifyAttackRange(UnitContext uc)
+  {
+    if( uc.rangeMax > 1 )
+      uc.rangeMax += rangeChange;
```

Comment:
If this only acts on units that already have a range that is greater than 1, is it impossible to buff Max artillery?

Reply:
Range 1 is direct combat. Max artillery have range 2-2.

Reply:
Ah, touché. And if it gets set less than `minRange`, `UnitContext.setWeapon` will sort that.